packages/providers/src/provider.ts

```diff
@@ -205,28 +207,75 @@ export type ProviderCallParams = {
   utxoValidation?: boolean;
 };
 
+/**
+ * URL - Consensus Params mapping.
+ */
+type ConsensusParamsCache = Record<string, ConsensusParameters>;
+
 /**
  * A provider for connecting to a node
  */
 export default class Provider {
   operations: ReturnType<typeof getOperationsSdk>;
   cache?: MemoryCache;
+  consensusParamsCache: ConsensusParamsCache = {};
 
   /**
    * Constructor to initialize a Provider.
    *
    * @param url - GraphQL endpoint of the Fuel node
+   * @param consensusParams - Consensus parameters of the Fuel node
    * @param options - Additional options for the provider
+   * @hidden
    */
   constructor(
     /** GraphQL endpoint of the Fuel node */
     public url: string,
+    consensusParams: ConsensusParameters,
     public options: ProviderOptions = {}
   ) {
     this.operations = this.createOperations(url, options);
+    this.consensusParamsCache[url] = consensusParams;
     this.cache = options.cacheUtxo ? new MemoryCache(options.cacheUtxo) : undefined;
   }
 
+  /**
+   * Creates a new instance of the Provider class. This is the recommended way to initialize a Provider.
+   * @param url - GraphQL endpoint of the Fuel node
+   * @param options - Additional options for the provider
+   */
+  static async connect(url: string, options: ProviderOptions = {}) {
+    const consensusParameters = await this.getConsensusParamsWithoutInstance(url);
+    const provider = new Provider(url, consensusParameters, options);
+    return provider;
+  }
+
+  /**
+   * Re-fetches the consensus parameters from the chain for the current URL.
+   */
+  async invalidateConsensusParamsCache() {
+    const { consensusParameters } = await this.getChain();
+    this.consensusParamsCache[this.url] = consensusParameters;
+  }
+
+  /**
+   * Returns the cached consensus parameters for the current URL.
+   */
+  getConsensusParams() {
+    return this.consensusParamsCache[this.url];
+  }
```

Comment:
this method is not needed if we use `getChain`

what will change is:
we need a new method to fetch from server
```
  async fetchChain(): Promise<ChainInfo> {
    const { chain } = await this.operations.getChain();
    return processGqlChain(chain);
  }

  // then getChain can be sync and handle cache / error
  getChain() {
    const chain = this.chainCache[this.url];
    if (!chain) throw new Error('Provider is not connected, please connect() it first.');
    return chain;
  }
```